featuretools/feature_discovery/feature_discovery.py

```diff
@@ -0,0 +1,571 @@
+import concurrent.futures
+import functools
+import inspect
+from itertools import combinations, permutations, product
+from typing import Callable, Dict, Iterable, List, Set, Tuple, Type, Union, cast
+
+import woodwork.type_sys.type_system as ww_type_system
+from woodwork.column_schema import ColumnSchema
+from woodwork.logical_types import LogicalType
+from woodwork.table_schema import TableSchema
+
+from featuretools.feature_discovery.type_defs import ANY, Feature, FeatureCollection
+from featuretools.primitives.base.primitive_base import PrimitiveBase
+from featuretools.tests.testing_utils.generate_fake_dataframe import flatten_list
+
+
+def generate_hashing_keys_from_column_schema(column_schema: ColumnSchema) -> List[str]:
+    """
+    Generate hashing keys from Columns Schema. For example:
+    - ColumnSchema(logical_type=Double) -> ["Double"]
+    - ColumnSchema(semantic_tags={"index"}) -> ["index"]
+    - ColumnSchema(logical_type=Double, semantic_tags={"index", "other"}) -> ["Double,index", "Double,other"]
+
+    Args:
+        column_schema (ColumnSchema):
+
+    Returns:
+        List[str]
+            List of hashing keys
+    """
+    keys: List[str] = []
+    logical_type = column_schema.logical_type
+    tags = column_schema.semantic_tags
+    lt_key = None
+    if logical_type:
+        lt_key = type(logical_type).__name__
+
+    if len(tags) > 0:
+        for tag in tags:
+            if tag is not None:
+                keys.append(f"{lt_key},{tag}" if lt_key is not None else tag)
```

Comment:
Interesting case. If ColumnA is defined as `ColumnSchema(logical_type=Double, semantic_tags={"index", "other"})` would that mean that it only fits input Features that match that signature exactly? 

So would we NOT want Feature(Double, ("index")) to match since it doesn't have the "other" tag?

Reply:
I think the question I'm trying to pose is how would we handle a primitive that required two semantic tags and a logical type for one of its inputs?

Reply:
yeah, I think you're right. I was trying to find an existing primitive that does this, but I don't think it exists. In the OR case, the primitive would just have a different line item, but I think I do need to think about this AND case, and your suggestion is right.

I think that means I only need 1 key for `"{lt_key},{tag1},{tag2}"` and multiple keys like I'm doing now

Reply:
refactored this keying to be part of the featurecollection creation